lightning: update docs for duplicate detection (#14343)
pingcap/docs: tidb-lightning/tidb-lightning-configuration.md

# will leak metadata about the data source.
# keep-after-success = false

[conflict]

> # The strategy to handle conflicting data

Reviewer: ```suggestion
# The strategy to handle conflicting data. The default value is "".
```

> # The strategy to handle conflicting data. The default value is "".

Reviewer: ```suggestion
# The new version strategy to handle conflicting data. The default value is "".
```

> # - "": no actions. TiDB Lightning might exit due to an error.

Reviewer: ```suggestion
# - "": TiDB Lightning does not detect and process conflicting dat. But if there are conflicting primary or unique key records in the source file, an error is reported in the subsequent step (Checksum).
```

> # - "error": the import task fails if there is conflicting data.

Reviewer: ```suggestion
# - "error": terminate the import and report an error if a primary or unique key conflict is detected in the imported data.
```

> # - "error": the import task fails if there is conflicting data.
> # - "replace": keep the new data when detecting conflicting data.
> # - "ignore": keep the old data when detecting conflicting data.

Reviewer: ```suggestion
# - "error": terminate the import and report an error if a primary or unique key conflict is detected in the imported data.
# - "replace": when encountering data with conflicting primary or unique keys, the new data is retained and the old data is overwritten.
# - "ignore": when encountering data with conflicting primary or unique keys, the old data is retained and the new data is ignored.
```

> # The strategy to handle conflicting data
> # - "": No actions, lightning may exit by error in following steps.
> # - "error": Conflicting data will fail the import task.
> # - "replace": Try to keep latest data when meet conflicting data.
> # - "ignore": Try to keep oldest data when meet conflicting data.
> strategy = ""
> # When `strategy` is "replace" or "ignore", controls the maximum tolerable conflict count. Only can be set when `strategy` is "replace" or "ignore". The default value is 9223372036854775807
> # threshold = 9223372036854775807
> # Controls the maximum record count in conflict_record table. The default value is 100
> # max-record-rows = 100

Reviewer: ```suggestion
# The strategy to handle conflicting data
# - "": no actions. TiDB Lightning might exit due to an error.
# - "error": the import task fails if there is conflicted data.
# - "replace": keep the new data when detecting conflicted data.
# - "ignore": keep the old data when detecting conflicted data.
strategy = ""
# When `strategy` is "replace" or "ignore", this parameter controls the upper limit of the conflicted data. You can set it only when `strategy` is "replace" or "ignore". The default value is 9223372036854775807.
# threshold = 9223372036854775807
# Controls the maximum number of rows in the `conflict_record` table. The default value is 100.
# max-record-rows = 100
```

> # Controls the maximum number of rows in the `conflict_record` table. The default value is 100.

Author: ```suggestion
# Controls the maximum number of rows in the `conflict_records` table. The default value is 100.
```

> # Controls the maximum number of rows in the `conflict_records` table. The default value is 100.

Reviewer: ```suggestion
# Controls the maximum number of rows in the `conflict_records` table. The default value is 100. If the strategy is "ignore", the conflict records that are ignored will be recorded; if the strategy is "replace", the conflict records that are overwritten will be recorded. However, there is a limitation that the replace strategy cannot record the conflict records in the logical import mode.
```

# will leak metadata about the data source.
# keep-after-success = false

[conflict]
# The new version strategy to handle conflicting data. The default value is "".

> # - "": TiDB Lightning does not detect and process conflicting dat. But if there are conflicting primary or unique key records in the source file, an error is reported in the subsequent step (Checksum).

Reviewer 2: ```suggestion
# - "": TiDB Lightning does not detect and process conflicting data. But if there are conflicting primary or unique key records in the source file, an error is reported in the subsequent step (Checksum).
```

> # - "": TiDB Lightning does not detect and process conflicting data. But if there are conflicting primary or unique key records in the source file, an error is reported in the subsequent step (Checksum).

Reviewer: ```suggestion
# - "": TiDB Lightning does not detect and process conflicting data. But if there are conflicting primary or unique key records in the source file, an error is reported in the subsequent step.
```

> # - "error": terminate the import and report an error if a primary or unique key conflict is detected in the imported data.
> # - "replace": when encountering data with conflicting primary or unique keys, the new data is retained and the old data is overwritten.
> # - "ignore": when encountering data with conflicting primary or unique keys, the old data is retained and the new data is ignored.

Reviewer: ```suggestion
# - "error": terminate the import and report an error if data with conflicting primary or unique keys is detected in the imported data.
# - "replace": when encountering data with conflicting primary or unique keys, the new data is retained and the old data is overwritten.
# - "ignore": when encountering data with conflicting primary or unique keys, the old data is retained and the new data is ignored.
```

# The new version strategy cannot be used together with tikv-importer.duplicate-resolution (the old version of conflict detection).
strategy = ""

> # Controls the upper limit of the conflicting data that can be handled when strategy is "replace" or "ignore". You can set it only when strategy is "replace" or "ignore". The default value is 9223372036854775807, which means that almost all errors are tolerated.

Reviewer 3: ```suggestion
# Controls the upper limit of the conflicting data that can be handled when strategy is "replace" or "ignore". You can set it only when strategy is "replace" or "ignore". The default value is 9223372036854775807, which means that almost all errors are tolerable.
```
